packages/valory/skills/decision_maker_abci/behaviours/storage_manager.py

```diff
@@ -308,11 +420,17 @@ def _set_policy(self) -> Generator:
             self._policy = self.synchronized_data.policy
             local_tools = self.synchronized_data.available_mech_tools
 
-        if self.is_first_period:
-            yield from self.wait_for_condition_with_sleep(
-                self._fetch_accuracy_info, sleep_time_override=self.params.sleep_time
-            )
-        self._update_accuracy_store(local_tools)
+        remote_accuracy_info = yield from self._fetch_accuracy_info()
```

Comment:
Fetching may fail, why not keep the original retry-until-success logic that the `wait_for_condition_with_sleep` offers?